docs/technical-design/code_coverage.md

```diff
@@ -0,0 +1,78 @@
+# Code coverage reports
+
+## Overview
+
+Code coverage is a measure of the percentage of code that is exercised during testing. Detailed coverage reports can identify which specific lines of code are being tested. In order for coverage to be measured, the code must first be instrumented. Tools that instrument code effectively wrap it in functions that increment a counter when that code is run. Code coverage generators take those counts and turn them into reports about covered and uncovered lines of code.
+
```

Comment:
Is this also what feed the code coverage metrics in Code Climate? If so, I feel like that is worth mentioning here.

Reply:
Good call.  Added that bit later in the doc.